### PR TITLE
Fix catalog drop table for dual storage

### PR DESCRIPTION
**Pull request by vikrambohra**
## Summary

Currently dropTable in catalog assumes the cluster default storage as the storage for all tables that were created. That assumption is incorrect with the introduction of storage per table. This PR fixes that bug/assumption by retrieving the correct fileio for a table being dropped and perform cleanup with the correct fileio

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [x] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [x] Refactoring
- [ ] Documentation
- [x] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [x] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

Added a DualStorageTest that tests creation and deletion of tables on hdfs and local.
# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: services/tables/src/main/java/com/linkedin/openhouse/tables/model/TableDto.java [hunk 1]
```diff
@@ -104,4 +104,13 @@ public Map<String, String> convertToMap() {
   private boolean isTableDtoPrimitive(Field field) {
     return field.getType() == String.class || field.getType() == long.class;
   }
+
+  /**
+   * Returns the primary key for TableDto Class
+   *
+   * @return TableDtoPrimaryKey
+   */
+  public TableDtoPrimaryKey getPrimaryKey() {
```
Comment (HotSushi) [inline services/tables/src/main/java/com/linkedin/openhouse/tables/model/TableDto.java:113]: can you add a method in `TableMapper.toTableDtoPrimaryKey(tableDto)` thats the singular place where we do such conversions

File: services/tables/src/test/java/com/linkedin/openhouse/tables/e2e/h2/DualStorageTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  @SpyBean @Autowired OpenHouseInternalRepository openHouseInternalRepository;
+
+  @Autowired StorageManager storageManager;
+
+  @Autowired Catalog catalog;
+
+  @Test
+  public void testCreateDropTableDualStorage() {
+
```
Comment (HotSushi) [inline services/tables/src/test/java/com/linkedin/openhouse/tables/e2e/h2/DualStorageTest.java:38]: nice test case ❤️

**Review by HotSushi — CHANGES_REQUESTED** [review]
single comment, otherwise looking good

File: services/tables/src/main/java/com/linkedin/openhouse/tables/model/TableDto.java (hunk 1 above)
Comment (vikrambohra) [reply services/tables/src/main/java/com/linkedin/openhouse/tables/model/TableDto.java]: Nice point. Made the change

**Review by HotSushi — APPROVED** [review]
LGTM

**Pull request merged**